src/components/application_manager/test/commands/mobile/add_command_request_test.cc

```diff
@@ -0,0 +1,299 @@
+    smart_objects::SmartObject menu_params =
+        smart_objects::SmartObject(smart_objects::SmartType_Map);
+
+    menu_params[strings::position] = 10;
+    menu_params[strings::menu_name] = "LG";
+    smart_objects::SmartObject msg_params =
+        smart_objects::SmartObject(smart_objects::SmartType_Map);
+    msg_params[strings::cmd_id] = 1;
+    msg_params[strings::app_id] = 34;
+    (*msg)[strings::msg_params] = msg_params;
+
+    return msg;
+  }
+};
+
+const hmi_apis::FunctionID::eType kInvalidFunctionId_ =
+    hmi_apis::FunctionID::INVALID_ENUM;
+
+TEST_F(AddCommandRequestTest, GetRunMethods_SUCCESS) {
+  MessageSharedPtr msg = create_fullParams_ui_so();
+  using namespace application_manager::commands;
+  utils::SharedPtr<AddCommandRequest> req =
+      CreateCommand<AddCommandRequest>(msg);
+
+  MockAppPtr mock_app = CreateMockApp();
+  ON_CALL(app_mngr_, application(_)).WillByDefault(Return(mock_app));
+
+  EXPECT_CALL(*mock_app, FindCommand(_));
+  EXPECT_CALL(*mock_app, AddCommand(_, _));
+
+  MockMessageHelper* mock_message_helper =
+      MockMessageHelper::message_helper_mock();
+  ON_CALL(*mock_message_helper, VerifyImage(_, _, _))
+      .WillByDefault(Return(mobile_apis::Result::SUCCESS));
+
+  application_manager::CommandsMap commands_map;
+  ON_CALL(*mock_app, commands_map())
+      .WillByDefault(Return(
+          DataAccessor<application_manager::CommandsMap>(commands_map, lock_)));
+
+  req->Run();
+}
+
+TEST_F(AddCommandRequestTest, OnEvent_UI_SUCCESS) {
+  MessageSharedPtr msg_ui = create_fullParams_ui_so();
+  using namespace application_manager::commands;
+  utils::SharedPtr<AddCommandRequest> req_ui =
+      CreateCommand<AddCommandRequest>(msg_ui);
+
+  MockAppPtr mock_app = CreateMockApp();
+  ON_CALL(app_mngr_, application(_)).WillByDefault(Return(mock_app));
+  ON_CALL(*mock_app, app_id()).WillByDefault(Return(1));
+
+  MessageSharedPtr msg = CreateMessage(smart_objects::SmartType_Map);
+  (*msg)[strings::params][hmi_response::code] =
+      hmi_apis::Common_Result::SUCCESS;
+  (*msg)[strings::msg_params][strings::cmd_id] = 1;
+
+  am::event_engine::Event event(hmi_apis::FunctionID::UI_AddCommand);
+  event.set_smart_object(*msg);
+
+  smart_objects::SmartObject* ptr = NULL;
+  ON_CALL(*mock_app, FindCommand(_)).WillByDefault(Return(ptr));
+
+  MockMessageHelper* mock_message_helper =
+      MockMessageHelper::message_helper_mock();
+  ON_CALL(*mock_message_helper, HMIToMobileResult(_))
+      .WillByDefault(Return(mobile_apis::Result::SUCCESS));
+
+  EXPECT_CALL(*mock_app, AddCommand(_, _));
+  EXPECT_CALL(*mock_app, UpdateHash());
+
+  am::CommandsMap commands_map;
+  ON_CALL(*mock_app, commands_map())
+      .WillByDefault(
+          Return(DataAccessor<am::CommandsMap>(commands_map, lock_)));
+
+  req_ui->Run();
+  req_ui->on_event(event);
+}
+
+TEST_F(AddCommandRequestTest, OnEvent_VR_SUCCESS) {
+  MessageSharedPtr msg_vr = create_fullParams_vr_so();
+  using namespace application_manager::commands;
```

Comment:
@wolfylambova22, you are using it at every test and just to get `AddCommandRequest`. Don't it would be better to just define usage of `AddCommandRequest` at the beginning of the file?